types/append_value.go

```diff
@@ -132,8 +132,14 @@ func ptrAppenderFunc(typ reflect.Type) AppenderFunc {
 }
 
 func appendValue(b []byte, v reflect.Value, flags int) []byte {
-	if v.Kind() == reflect.Ptr && v.IsNil() {
-		return AppendNull(b, flags)
+	switch v.Kind() {
+	case reflect.Ptr, reflect.Slice, reflect.Map:
+		// When the value is a pointer, slice or map; then we can assert if the value provided is nil.
+		// When the value is properly nil; then we can append NULL instead of a string representation of the provided
+		// value.
+		if v.IsNil() {
+			return AppendNull(b, flags)
+		}
 	}
 	appender := Appender(v.Type())
```

Comment:
The idea is that you should be able to call `Appender` once for a type and then use it to append different values of the given type (nil or non-nil). So the function returned by appender should handle nil values, not `appendValue` itself.

I guess it is not worth fixing this properly so let's keep this code, but at least we should have an end-to-end test that inserts and selects slice/map from a real table...